src/precord.c

```diff
@@ -818,35 +826,37 @@ Obj FuncLT_PREC (
     SortPRecRNam(left,0);
     SortPRecRNam(right,0);
 
+    CheckRecursionBefore();
+    res = 0;
+
     /* compare componentwise                                               */
     for ( i = 1; i <= LEN_PREC(right); i++ ) {
 
         /* if the left is a proper prefix of the right one                 */
-        if ( LEN_PREC(left) < i )  return True;
+        if ( LEN_PREC(left) < i ) {
+            res = 1;
+            break;
+        }
 
         /* compare the names                                               */
         /* The sense of this comparison is determined by the rule that
            unbound entries compare less than bound ones                    */
         if ( GET_RNAM_PREC(left,i) != GET_RNAM_PREC(right,i) ) {
-            if ( strcmp( NAME_RNAM( labs((Int)(GET_RNAM_PREC(left,i))) ),
-                   NAME_RNAM( labs((Int)(GET_RNAM_PREC(right,i))) ) ) > 0 ) {
-                return True;
-            }
-            else {
-                return False;
-            }
+            res = ( strcmp( NAME_RNAM( labs((Int)(GET_RNAM_PREC(left,i))) ),
+                   NAME_RNAM( labs((Int)(GET_RNAM_PREC(right,i))) ) ) > 0 );
+            break;
         }
 
         /* compare the values                                              */
         if ( ! EQ(GET_ELM_PREC(left,i),GET_ELM_PREC(right,i)) ) {
-            return LT( GET_ELM_PREC(left,i), GET_ELM_PREC(right,i) ) ?
-                   True : False;
+            res = LT( GET_ELM_PREC(left,i), GET_ELM_PREC(right,i) );
+            break;
         }
 
     }
 
     /* the records are equal or the right is a prefix of the left          */
-    return False;
```

Comment:
Absolutely. This is one place where I'd love we were in C++ and could manage this via RAII